lib/iron_bank/user.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module IronBank
+  # A Zuora user, only available if the user data has been exported and provided
+  # to IronBank through the `zuora_users_file` configuration option.
+  #
+  # Cf. https://knowledgecenter.zuora.com/CF_Users_and_Administrators/A_Administrator_Settings/Manage_Users#Export_User_Data_to_a_CSV_File
+  #
+  class User
+    extend IronBank::Local
+
+    class << self
+      def store
+        zuora_users_file = IronBank.configuration.zuora_users_file
+        return {} unless zuora_users_file
+
+        @store ||= begin
+          if File.exist?(zuora_users_file)
+            load_records
+          else
+            IronBank.logger.warn "File does not exist: #{zuora_users_file}"
+            {}
+          end
+        end
+      end
+
+      def load_records
+        file = IronBank.configuration.zuora_users_file
+
+        CSV.foreach(file, headers: true).with_object({}) do |row, store|
+          store[row["User ID"]] = new(row.to_h.compact)
+        end
+      end
+
+      def all
+        store.values
+      end
+
+      def find(user_id)
+        store[user_id]
+      end
+    end
+
+    US_DATE_FORMAT = "%m/%d/%Y"
+
+    FIELDS = [
```

Comment:
Are we protected against Zuora adding new fields?

Reply:
yes